manifests/database/postgresql.pp

```diff
@@ -9,12 +9,14 @@
   $manage_server        = $puppetdb::params::manage_dbserver,
   $manage_package_repo  = $puppetdb::params::manage_pg_repo,
   $postgres_version     = $puppetdb::params::postgres_version,
+  $postgres_datadir     = $puppetdb::params::postgres_datadir,
```

Comment:
Its because class `puppetdb` also configures postgresql, but if instead you use `puppetdb::server` it decomposes `puppetdb` and doesn't pull in postgresql:

```
class { 'puppetdb::server':
  listen_address => '0.0.0.0',
}

class { ':postgresql::globals':
  manage_package_report => true,
  version => '9.4',
}

class { '::postgresql::server':
  datadir => '/srv/pg_data',
}

postgresql::server::extension { 'pg_trgm':
  database  => 'puppetdb',
}

postgresql::server::db { $database_name:
  user     => 'puppetdb',
  password => 'mypassword',
  grant    => 'all',
}
```

Another decomposition option, is to pass in manage_server => false, which doesn't manage postgresql, but creates the database and user for you:

```
class { 'puppetdb':
  listen_address => '0.0.0.0',
  manager_dbserver => false,
}

class { ':postgresql::globals':
  manage_package_report => true,
  version => '9.4',
}

class { '::postgresql::server':
  datadir => '/srv/pg_data',
}

postgresql::server::extension { 'pg_trgm':
  database  => 'puppetdb',
}
```